src/Raven.Server/Documents/PeriodicBackup/Restore/RestoreBackupTask.cs

```diff
@@ -545,13 +553,32 @@ private List<string> GetFilesForRestore(string backupLocation)
             }
         }
 
-        private Stream GetInputStream(FileStream fileStream)
+        private Stream GetSnapshotInputStream(Stream fileStream, string database)
         {
-            if (_restoreConfiguration.EncryptionSettings == null)
-                return fileStream;
+            using (_serverStore.ContextPool.AllocateOperationContext(out TransactionOperationContext ctx))
+            using (ctx.OpenReadTransaction())
+            {
+                if (_restoreConfiguration.BackupEncryptionSettings?.Key != null)
+                {
+                    return new DecryptingXChaCha20Oly1305Stream(fileStream, Convert.FromBase64String(_restoreConfiguration.BackupEncryptionSettings.Key));
+                }
+                var key = _serverStore.GetSecretKey(ctx, database);
+                if (key != null)
```

Comment:
Doesn't this mean that encrypted db always restore from encrypted database?
But we are restoring the db at this point, so I don't see how this can be possible? Where is that key coming from?

Reply:
In https://github.com/ravendb/ravendb/pull/8506 I changed this method.
The key is from _restoreConfiguration.EncryptionKey . we do  '_serverStore.PutSecretKey(_restoreConfiguration.EncryptionKey,...' in Execute

Reply:
So we need to check on `FileStream` and then call `Flush(flushToDisk: true)` here, no?